cmd/flat-network-controller/app/controller.go

```diff
@@ -0,0 +1,247 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	v1 "k8s.io/arktos-ext/pkg/apis/arktosextensions/v1"
+	arktos "k8s.io/arktos-ext/pkg/generated/clientset/versioned"
+	arktoscheme "k8s.io/arktos-ext/pkg/generated/clientset/versioned/scheme"
+	arktosinformer "k8s.io/arktos-ext/pkg/generated/informers/externalversions/arktosextensions/v1"
+	arktosv1 "k8s.io/arktos-ext/pkg/generated/listers/arktosextensions/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+	"strings"
+	"time"
+)
+
+// Controller represents the flat network controller
+type Controller struct {
+	cacheSynced  cache.InformerSynced
+	store        arktosv1.NetworkLister
+	queue        workqueue.RateLimitingInterface
+	netClientset *arktos.Clientset
+	svcClientset *kubernetes.Clientset
+	recorder     record.EventRecorder
+}
+
+// New creates the controller object
+func New(netClientset *arktos.Clientset, svcClientset *kubernetes.Clientset, informer arktosinformer.NetworkInformer) *Controller {
+	utilruntime.Must(arktoscheme.AddToScheme(scheme.Scheme))
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(klog.Infof)
+	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: svcClientset.CoreV1().EventsWithMultiTenancy(metav1.NamespaceAll, metav1.TenantAll)})
+
+	return &Controller{
+		store:        informer.Lister(),
+		queue:        workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		cacheSynced:  informer.Informer().HasSynced,
+		netClientset: netClientset,
+		svcClientset: svcClientset,
+		recorder: eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "flat-network-controller"}),
+	}
+}
+
+// Run starts the control loop with workers processing the items
+func (c *Controller) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Info("starting flat network controller")
+	klog.V(5).Info("waiting for informer caches to sync")
+	if !cache.WaitForCacheSync(stopCh, c.cacheSynced) {
+		klog.Error("failed to wait for cache to sync")
+		return
+	}
+
+	klog.V(5).Info("staring workers of flat network controller")
+	for i := 0; i < workers; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+
+	klog.V(5).Infof("%d workers started", workers)
+	<-stopCh
+	klog.Info("shutting down flat network controller")
+}
+
+func (c *Controller) runWorker() {
+	for {
+		item, shutdown := c.queue.Get()
+		if shutdown {
+			break
+		}
+
+		c.process(item)
+	}
+}
+
+// processItem will read a single work item off the work queue and attempt to process it
+func (c *Controller) process(item interface{}) {
```

Comment:
comment says "processItem", but func name is "process". I think processNextWorkItem is better, which is consistent with other controller/operator code in K8s.

Reply:
will correct in comment. process(item), if read method together with parameter, it is process item already.